src/coreclr/vm/stubgen.cpp

```diff
@@ -2315,17 +2336,39 @@ static BOOL SigHasVoidReturnType(const Signature &signature)
     return (ELEMENT_TYPE_VOID == retType);
 }
 
+namespace
```

Comment:
Can you remind me what is the advantage of `namespace` with extra indentation vs. just using `static`?

This rest of this file is using `static`, so this leads to inconsistent style.

FWIW, I see the extra indentation as nuisance. I am happy that C# is working on introducing language features to make the indentation for namespaces unnecessary.

Reply:
>Can you remind me what is the advantage of namespace with extra indentation vs. just using static?

The end result is the same, but the advantage is avoiding the `static` overuse of the keyword and more or less the same as any other namespace - logical grouping. Having disparate functions with `static` doesn't convey they may be a group or collection. There are clearly other ways to accomplish this but that is why I use it - "grouping" and leaving `static` for applying to a member.

Something to consider when using a anonymous namespaces is that it can make debugging harder. Since a namespace is actually being introduced debugging types when one is outside of the translation unit can be annoying. A type in an anonymous namespace may not be directly usable during function evals since the debugger can have difficulty in resolving it without an indication to look inside of these namespaces. See [here](https://developercommunity.visualstudio.com/idea/388773/support-anonymous-namespaces-in-the-debugger.html) for an example in Visual Studio.

>This rest of this file is using static, so this leads to inconsistent style.

Agree. We should style consistently within a file - ideally across the code base IMO.

Reply:
Do you know of a project with written down C++ coding conventions that say to prefer anonymous namespaces over static keyword?

LLVM coding conventions that I like for its level of detail and taste say to use anonymous namespaces for classes only: https://llvm.org/docs/CodingStandards.html#anonymous-namespaces

Reply:
>Do you know of a project with written down C++ coding conventions that say to prefer anonymous namespaces over static keyword?

The convention is a part of the C++ Core Guidelines defined [here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf22-use-an-unnamed-anonymous-namespace-for-all-internalnon-exported-entities). The primary point of the anonymous namespace was to reduce the overreliance on the `static` keyword - therefore the argument is "use C++ features instead of inherited C features". Google on the other hand [doesn't take a strong stance](https://google.github.io/styleguide/cppguide.html#Unnamed_Namespaces_and_Static_Variables).

I personally prefer to use the appropriate C++ mechanism when available but style almost always overrides that preference. I completely understand the indentation argument so am sympathetic. The argument presented in LLVM is a pragmatic view and also is reasonable, but again C++ has a very clear mechanism for this and avoids the ambiguity present when one observes the `static` keyword - if one is familiar with C/C++ it is clear but coming from C# or Java it can be confusing since it serves no functional purpose and is more a "linker directive".

I am all for keeping the current style in the translation unit so `static` works for me. The [`interop` library](https://github.com/dotnet/runtime/tree/master/src/coreclr/interop) defers to C++ mechanisms when available so has opted for using anonymous namespaces, but I am more than happy to say the entire dotnet/runtime code base follows LLVM style guidelines. If that is the case we should update our documentation.

Reply:
Regardless of our decision here, I have requested ISO/C++ guidance at https://github.com/isocpp/CppCoreGuidelines/issues/1727.

Reply:
@AaronRobinsonMSFT Thank you for following up on this!